docs/pages/machine-id/reference/configuration.mdx

```diff
@@ -0,0 +1,63 @@
+---
+title: Machine ID Configuration Reference
+description: Configuration reference for Teleport Machine ID.
+---
+
+```yaml
+# Debug enables verbose logging to stderr.
+debug: true
+
+# Address of the Teleport Auth Server (on-prem installs) or Proxy Server
+# (Cloud installs).
+auth_server: "auth.example.com:3025" (On-Prem)
+auth_server: "example.teleport.sh:443" (Cloud)
+
+# TTL of short-lived machine certificates.
+certificate_ttl: "5m"
+
+# Interval at which short-lived certificates are renewed; must be less than
+# the certificate TTL.
+renewal_interval: "1m"
+
+# If set, quit after the first renewal.
+oneshot: true
+
+# Onboarding values only used on first connect.
+onboarding:
+    # Cluster join method. Can be "token" or "iam".
+    join_method: "token"
+    # Token used to join cluster.
+    token: "00000000000000000000000000000000"
+    # CA Pin used to validate the identity of Teleport Auth Server on first connect.
+    ca_pin: "sha256:1111111111111111111111111111111111111111111111111111111111111111"
+    # CA Pins used to validate the identity of Teleport Auth Server on first connect.
+    ca_pins:
+    - "sha256:1111111111111111111111111111111111111111111111111111111111111111"
+    - "sha256:2222222222222222222222222222222222222222222222222222222222222222"
+
+# Storage defines where Machine ID internal data is stored.
+storage:
+  # Directory to store internal bot data. Access to this directory should be
+  # limited.
+  directory: /var/lib/teleport/bot
```

Comment:
If we're aiming to be exhaustive, it might be worth mentioning you can alternatively use in-memory storage too:

```yaml
storage:
  # Directory to store internal bot data. Access to this directory should be
  # limited.
  directory: /var/lib/teleport/bot

  # Alternatively, internal data can be stored in memory. Other storage types
  # (e.g. `directory`) should be commented out.
  # memory: true
```

(memory storage is probably appealing right now for non-IAM, one-shot bots since this will never write the renewable cert to disk)